File: 1614-nesting-depth/solution.go
package main

func maxDepth(s string) int {
	n := len(s)
	if n <= 1 && s != "(" && s != ")" {
		return 0
	}

	maxDepth := 0
	currentDepth := 0

	for _, c := range s {
		if c == '(' {
			currentDepth++
			if currentDepth > maxDepth {
				maxDepth = currentDepth
			}
		} else if c == ')' {
			currentDepth--
		}
	}
	return maxDepth
}

File: 1614-nesting-depth/solution.py
class Solution:
    def maxDepth(self, s: str) -> int:
        n = len(s)
        if n <= 1 and s not in ('(', ')'):
            return 0
        
        current_depth = 0
        max_depth = 0

        for c in s:
            if c == "(":
                current_depth += 1
                if current_depth > max_depth:
                    max_depth = current_depth
            elif c == ")":
                current_depth -= 1

        return max_depth